types/pubsub-js/index.d.ts

```diff
@@ -2,46 +2,53 @@
 // Project: https://github.com/mroderick/PubSubJS
 // Definitions by: Boris Yankov <https://github.com/borisyankov>
 //                 Matthias Lindinger <https://github.com/morpheus-87>
+//                 Profesor08 <https://github.com/Profesor08>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare namespace PubSubJS {
-    interface Base
+    interface Base<T = any, M = Message>
         extends CountSubscriptions,
             ClearAllSubscriptions,
             GetSubscriptions,
-            Publish,
-            Subscribe,
-            Unsubscribe {
+            Publish<T, M>,
+            Subscribe<T, M>,
+            Unsubscribe<T> {
         name: string;
         version: string;
     }
 
+    type Token = string;
+
+    type Message = string | Symbol;
+
+    type SubscriptionListener<T> = (message: Message, data?: T) => void;
+
     interface CountSubscriptions {
-        countSubscriptions(token: any): number;
+        countSubscriptions(token: Token): number;
     }
 
     interface ClearAllSubscriptions {
-        clearAllSubscriptions(token?: any): void;
+        clearAllSubscriptions(token?: Token): void;
     }
 
     interface GetSubscriptions {
-        getSubscriptions(token: any): any[];
+        getSubscriptions(token: Token): Message[];
     }
 
-    interface Publish {
-        publish(message: string | Symbol, data?: any): boolean;
+    interface Publish<T, M> {
```

Comment:
those `M` should extend `Message`?
what's the reason you're adding generics?
The change for 'Message' is unproblematic, those need some explanation (what's a gain)

Reply:
> @peterblazejewicz those `M` should extend `Message`?

Yes, this mistake if fixed.

> @peterblazejewicz what's the reason you're adding generics?

To have full control over the `data` type and to have code assistance and to make cool things. This will improve code quality. `any` type is unsafe. And any way generics are optional.

Reply:
this is not optional, existing code would break, e.g.:
```ts
 // create a subscriber to receive all messages from a hierarchy of topics
    var myToplevelSubscriber = (msg: string, data: any) => {
        console.log('top level: ', msg, data);
    };

    // subscribe to all topics in the 'car' hierarchy
    PubSub.subscribe('car', myToplevelSubscriber);
 ```
do you need to provide fallback, to make new improved types optional? Something like:
```ts
    type SubscriptionListener<T, TMessageLike extends Message = string> = (message: TMessageLike, data?: T) => void;
```

Reply:
> this is not optional IMO, existing code would break, e.g.:

It will, because current type definition for message is wrong.

Reply:
this is from current PR, if someone has a 'string' as per docs (msg: string), this would rise error now `Argument of type '(msg: string, data: any) => void' is not assignable to parameter of type 'SubscriptionListener<any>'`, the string is technically correct from docs, is it? 
My concern is, library documented 'string' as a message (mostly), while using 'symbol' in the details:
https://github.com/mroderick/PubSubJS/blob/master/src/pubsub.js
what is the possibility we break someone else code? I'm not owner of this types, never used this lib for pubsub implementation.

Reply:
https://github.com/mroderick/PubSubJS#example-of-use-of-symbol-constants-with-es67-syntax

Reply:
According to source it converts symbol to string. Added a fix.

Reply:
cool, now it's compatible with previous code.